src/eduid_webapp/phone/views.py

```diff
@@ -187,48 +195,29 @@ def verify(user, code, number):
             '_status': 'error',
             'message': 'phones.code_expired_send_new'
         }
-
     if code != state.verification.verification_code:
         msg = "Invalid verification code: {!r}".format(state.verification)
         current_app.logger.debug(msg)
         return {
             '_status': 'error',
             'message': 'phones.code_invalid'
         }
-
-    current_app.proofing_statedb.remove_state(state)
-
-    new_phone = PhoneNumber(number = number, application = 'eduid_phone',
-                            verified = True, primary = False)
-
-    has_primary = proofing_user.phone_numbers.primary
-    if has_primary is None:
-        new_phone.is_primary = True
-    try:
-        proofing_user.phone_numbers.add(new_phone)
-    except DuplicateElementViolation:
-        proofing_user.phone_numbers.find(number).is_verified = True
-        if has_primary is None:
-            proofing_user.phone_numbers.find(number).is_primary = True
-
     try:
-        save_and_sync_user(proofing_user)
+        verify_phone_number(state, proofing_user)
+        current_app.logger.info('phone number {!r} successfully verified'
+                                 ' for user {!r}'.format(number, proofing_user))
+        phones = {
+                'phones': proofing_user.phone_numbers.to_list_of_dicts(),
+                'message': 'phones.verification-success'
```

Comment:
All the other messages use underscore, while this one (and the same one for emails) uses hyphens - intentional?

Reply:
It's the string that is registered as msg id in the react app. There is no restriction, other than using a short string. There are some with spaces...